core/src/main/scala/org/apache/spark/scheduler/SparkListener.scala

```diff
@@ -121,6 +121,9 @@ case class SparkListenerApplicationStart(
 @DeveloperApi
 case class SparkListenerApplicationEnd(time: Long) extends SparkListenerEvent
 
+@DeveloperApi
+case class SparkListenerContextStop() extends SparkListenerEvent
```

Comment:
I'm not sure why you need this. Why won't `SparkListenerApplicationEnd` work? It's send in `SparkContext.stop()`.